### PR TITLE
feat(DEV-12179): custom Tabs for Payables

### PR DESCRIPTION
**Pull request by vkuprin**
Screenshot 2024-09-24 at 01 59 20

### CODE REVIEW

File: packages/sdk-react/src/components/payables/PayablesTable/Filters/SummaryCardsFilters.tsx [hunk 1]
```diff
@@ -73,7 +74,7 @@ const AmountTypography = styled(Typography)(() => ({
   alignItems: 'baseline',
 }));
 
-const className = 'Monite-SummaryCard';
+export const sumaryClassName = 'Monite-SummaryCard';
```
Comment (PeCrio) [inline packages/sdk-react/src/components/payables/PayablesTable/Filters/SummaryCardsFilters.tsx:77]:
```suggestion
export const summaryClassName = 'Monite-SummaryCard';
```

File: packages/sdk-react/src/components/payables/PayablesTable/consts.ts [hunk 2]
```diff
@@ -16,3 +16,15 @@ export const DEFAULT_FIELD_ORDER: FieldValueTypes[] = [
   'amount',
   'pay',
 ];
+
+export const DEFAULT_CARDS_ORDER = [
```
Comment (PeCrio) [inline packages/sdk-react/src/components/payables/PayablesTable/consts.ts:20]: Would be nice to have a type on this constant.

File: packages/sdk-themes/src/themes/monite.ts [hunk 3]
```diff
@@ -945,6 +953,24 @@ export const defaultMoniteComponents: Components<Omit<Theme, 'components'>> = {
         'amount',
         'pay',
       ],
+      tab_filters: {
```
Comment (saladNights) [inline packages/sdk-themes/src/themes/monite.ts:956]: mix of different Casings, let's stick to a single one. The theme object supports the camel case.
Or am I missing the reason for the snake case here?

**Review by saladNights — CHANGES_REQUESTED** [review]

File: packages/sdk-react/src/ui/DueDateCell/DueDateCell.tsx [hunk 4]
```diff
@@ -37,9 +37,10 @@ export const DueDateCell = ({ data }: DueDateCellProps) => {
           fontWeight="bold"
           fontSize="small"
         >
-          {t(i18n)`Overdue by ${overdueDays} ${
-            overdueDays === 1 ? 'day' : 'days'
-          }`}
+          {overdueDays > 0 &&
```
Comment (saladNights) [inline packages/sdk-react/src/ui/DueDateCell/DueDateCell.tsx:40]: There is a function called `plural` in Lingui. Check the example in the file `packages/sdk-react/src/components/receivables/InvoiceDetails/ExistingInvoiceDetails/components/reminderCardTermsHelpers.tsx`
Let's use a single approach to pluralize translations

File: packages/sdk-react/src/components/payables/PayablesTable/consts.ts [hunk 5]
```diff
@@ -4,7 +4,7 @@ export const FILTER_TYPE_SEARCH = 'search';
 export const FILTER_TYPE_STATUS = 'status';
 export const FILTER_TYPE_DUE_DATE = 'due_date';
 export const FILTER_TYPE_CREATED_AT = 'created_at';
-export const FILTER_TYPE_OVERDUE = 'overdue';
+export const FILTER_TYPE_CUSTOM_MONITE = 'custom_monite';
```
Comment (saladNights) [inline packages/sdk-react/src/components/payables/PayablesTable/consts.ts:7]: It was difficult to understand the purpose of this filter by name. Let's use something more descriptive. From what I understood it should be something like `summary_card_filter`

**Review by saladNights — CHANGES_REQUESTED** [review]

File: packages/sdk-themes/src/themes/monite.ts (hunk 3 above)
Comment (vkuprin) [reply packages/sdk-themes/src/themes/monite.ts:956]: Certainly, I initially used the format mentioned in the ticket based on our customers’ requirements. However, I believe it’s important to prioritize following established conventions. We can also communicate with the customers to ensure alignment with this approach

Comment (saladNights) [reply packages/sdk-themes/src/themes/monite.ts]: realized that tabFilters is a legacy from another task. In your case, it should be `cardFilters` or `summaryCardFilters`

File: packages/sdk-react/src/components/payables/PayablesTable/consts.ts [hunk 6]
```diff
@@ -1,10 +1,11 @@
+import { ExtendedPayableStateEnum } from '@/components/payables/PayablesTable/Filters/SummaryCardsFilters';
 import { FieldValueTypes } from '@/components/payables/PayablesTable/types';
 
 export const FILTER_TYPE_SEARCH = 'search';
 export const FILTER_TYPE_STATUS = 'status';
 export const FILTER_TYPE_DUE_DATE = 'due_date';
 export const FILTER_TYPE_CREATED_AT = 'created_at';
-export const FILTER_TYPE_OVERDUE = 'overdue';
+export const FILTER_TYPE_CUSTOM_MONITE = 'summary_card_filter';
```
Comment (saladNights) [inline packages/sdk-react/src/components/payables/PayablesTable/consts.ts:8]:
```suggestion
export const FILTER_TYPE_SUMMARY_CARD = 'summary_card_filter';
```

**Review by saladNights — CHANGES_REQUESTED** [review]

**Review by saladNights — APPROVED** [review]

**Review by PeCrio — APPROVED** [review]

**Pull request merged**